polaris-react/src/components/IndexTable/IndexTable.tsx

```diff
@@ -440,7 +451,7 @@ function IndexTableBase({
     scrollingContainer.current = false;
   }, []);
 
-  useEffect(() => {
+  useIsomorphicLayoutEffect(() => {
```

Comment:
I believe by changing this to an `useIsomorphicLayoutEffect`, this will run after the initial paint and thus be able to run the `resizeTableHeadings` with the proper widths.

@kyledurand @mattkubej can you tophat this fix? The spin link should be updated with the latest snapshot and changes.

Reply:
Oh nice find! This does seem to fix it

Reply:
This makes sense. I had to do the same thing for page actions that did some measuring after web shipped react 18